src/components/TablePagination/index.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { UsePaginationInstanceProps, UsePaginationState } from 'react-table';
+import classnames from 'classnames';
+
+import './index.scss';
+
+type ReactTablePaginationProps = {
+  className?: string;
+} & UsePaginationInstanceProps<{}> &
+  UsePaginationState<{}>;
+
+const getVisiblePages = (page: number, total: number) => {
+  if (total < 7) {
+    return [1, 2, 3, 4, 5].filter(singlePage => singlePage <= total);
+  }
+  if (page % 4 >= 0 && page > 3 && page + 2 < total) {
+    return [1, page - 1, page, page + 1, total];
+  }
+  if (page % 4 >= 0 && page > 3 && page + 2 >= total) {
+    return [1, total - 3, total - 2, total - 1, total];
+  }
+  return [1, 2, 3, 4, total];
+};
+
+const TablePagination = ({
+  className,
+  gotoPage,
+  previousPage,
+  nextPage,
+  canNextPage,
+  pageIndex,
+  pageOptions,
+  canPreviousPage
+}: ReactTablePaginationProps) => {
+  const { t } = useTranslation('systemProfile');
+  const [visiblePages, setVisiblePages] = useState<number[]>([]);
```

Comment:
I don't think this hook is needed (or the `useEffect()` call that calls `setVisiblePages`). Down on line 72, instead of using `visiblePages.map(...)`, can we just use `getVisiblePages(pageIndex + 1, pageOptions.length).map(...)`?

Reply:
Yep that can be done 👍